app/src/components/main/modal/modalLayout/modalFooter/modalFooter.jsx

```diff
@@ -95,14 +102,25 @@ export class ModalFooter extends Component {
             <div className={cx('button-container')}>
               <BigButton
                 color={okButton.danger ? 'tomato' : 'booger'}
-                onClick={onClickOk}
+                onClick={() => okButton.onClick(onClickOk)}
                 disabled={okButton.disabled}
               >
                 {okButton.text}
               </BigButton>
             </div>
           )}
-          {customButton ? <div className={cx('button-container')}>{customButton}</div> : null}
+          {customButton && (
+            <div className={cx('button-container')}>
+              {customButton.component ? (
+                <customButton.component
+                  {...customButton.buttonProps}
+                  onClick={() => customButton.onClick(onClickOk)}
```

Comment:
`onClick={() => customButton.onClick(closeHandler)}`

Reply:
For now closeHandler run with confirmation logic. For okButton we don't need to do it.